### PR TITLE
Add: Evil Support

### PR DESCRIPTION
**Pull request by larebsyed**
Adding evil support, 
Requirese `evil-search-module` to be `isearch`

Fix: #33

### CODE REVIEW

**Comment by larebsyed** [comment]
@tam17aki This is my first try, and there might be something missing. 
I am testing this in my own setup. 
Please take a look and let me know if you think this can work.

**Comment by larebsyed** [comment]
bump @tam17aki

**Comment by tam17aki** [comment]
Thank you very much for pull-req.
I will check the contents in this week.

**Comment by tam17aki** [comment]
I have tested the PR on my environment (Emacs 27.1, MacOS Catalina).
It works well. I'll test it some more.
Thank you very much!

File: ace-isearch.el [hunk 1]
```diff
@@ -218,16 +218,15 @@ of `isearch-string' is longer than or equal to `ace-isearch-input-length'."
            2)))
     (cond (;; using avy/ace-jump since L=1 or L=2 reached (depending on `ace-isearch-jump-based-on-one-char')
            (and (= (length isearch-string) ace-isearch-input-min-length)
-                (and (not isearch-regexp)
-                     (or (not (ace-isearch--isearch-regexp-function))
+                (and (or (not (ace-isearch--isearch-regexp-function))
```
Comment (tam17aki) [inline ace-isearch.el:221]: Good fix. 
By introducing the variable below, 
```elisp
(defcustom ace-isearch-on-evil-mode nil
  "If non nil, ace-isearch-mode can be used on Evil mode."
  :type 'boolean
  :group 'ace-isearch)
```
the code can be fixed as follows:

```elisp
 (and (if ace-isearch-on-evil-mode
         t
       (not isearch-regexp))
     (or (not (ace-isearch--isearch-regexp-function))
         (not (eq search-default-mode nil))))
```

File: ace-isearch.el [hunk 2]
```diff
@@ -218,16 +218,15 @@ of `isearch-string' is longer than or equal to `ace-isearch-input-length'."
...
                          (not (eq search-default-mode nil))))
                 (ace-isearch--fboundp (if ace-isearch-jump-based-on-one-char
-                                          ace-isearch-function ace-isearch-2-function)
+                                         ace-isearch-function ace-isearch-2-function)
```
Comment (tam17aki) [inline ace-isearch.el:224]: uncessesary.

File: ace-isearch.el [hunk 3]
```diff
@@ -251,11 +250,10 @@ of `isearch-string' is longer than or equal to `ace-isearch-input-length'."
 
           ;; switching from isearch to helm/swiper since `ace-isearch-input-length' reached
           ((and (>= (length isearch-string) ace-isearch-input-length)
-                (not isearch-regexp)
```
Comment (tam17aki) [inline ace-isearch.el:254]: I think this can be fixed as
```elisp
(if ace-isearch-on-evil-mode
    t
  (not isearch-regexp))
```

**Comment by larebsyed** [comment]
@tam17aki Thank you for the review. I have been using this for a while, there is only one issue that I am facing is that when we search a string that is long enough for swiper. The search string is not stored for a repeating the search. Do you have any idea how to fix that?

**Comment by tam17aki** [comment]
please let me confirm: have you set (setq ace-isearch-on-evil-mode t) ?

**Comment by larebsyed** [comment]
@tam17aki Not yet, will do over the weekend. sorry for delays.

**Comment by larebsyed** [comment]
@tam17aki Did the requested changes, and also fixed the issue. Please review when you can.

File: ace-isearch.el [hunk 4]
```diff
@@ -227,7 +234,7 @@ of `isearch-string' is longer than or equal to `ace-isearch-input-length'."
                       (and (eq ace-isearch-use-jump 'printing-char)
                            (eq this-command 'isearch-printing-char))))
                 (sit-for ace-isearch-jump-delay))
-           (isearch-exit)
+           (isearch-done nil t)
```
Comment (larebsyed) [inline ace-isearch.el:237]:
```suggestion
           (isearch-done t t)
```

**Comment by tam17aki** [comment]
Thank you!
I'll use it for a while to check it out.
If there are no problems identified, I will merge it into the master branch.

**Pull request merged**